pkg/analyze/kube_resource.go

```diff
@@ -0,0 +1,172 @@
+package analyzer
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strconv"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	iutils "github.com/replicatedhq/troubleshoot/pkg/interfaceutils"
+	"gopkg.in/yaml.v2"
+)
+
+func FindResource(kind string, namespace string, name string, getFileContents func(string) (map[string][]byte, error)) (interface{}, error) {
```

Comment:
> What if I have multiple objects with the same name in different namespaces?
it shouldn't matter since to select a namespaced object you have to be explicit about the namespace

Reply:
> it shouldn't matter since to select a namespaced object you have to be explicit about the namespace

I could run an analyser without selecting a namespace.

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: example
spec:
  collectors:
    - cluster-resources: {}
  analyzers:
    - clusterResource:
        checkName: Nginx image version
        kind: Pod
        name: nginx
        yamlPath: "spec.containers.[0].image"
        value: nginx:1.23
        outcomes:
            - fail:
                when: "false"
                message: The nginx version is old
            - pass:
                when: "true"
                message: Happy that you have the latest nginx version
```

I have two pods, one with an old version and the other with a new version

```sh
[evans] $ k get pods -A | grep nginx
default       nginx                                     1/1     Running     0                 4m23s       
kube-system   nginx                                     1/1     Running     0                 4m23s  
```

When I analyse a support bundle generated using the spec above I get a pass. The analyser picks the pod in the kube-system namespace (`nginx:1.23`). The one in the default is `nginx:1.23`

Reply:
To run the analyser spec above I had to apply [this](https://github.com/replicatedhq/troubleshoot/pull/780#discussion_r1026310814) patch first, else an error was just being raised.